### PR TITLE
stop hardcoding pk

### PR DESCRIPTION
**Pull request by edgao**

### CODE REVIEW

File: airbyte-integrations/connectors/destination-bigquery/src/test-integration/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGeneratorIntegrationTest.java [hunk 1]
```diff
@@ -10,8 +10,19 @@
...
+import com.google.cloud.bigquery.DatasetInfo;
+import com.google.cloud.bigquery.Field;
 import com.google.cloud.bigquery.Field.Mode;
+import com.google.cloud.bigquery.FieldValue;
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import com.google.cloud.bigquery.Table;
+import com.google.cloud.bigquery.TableResult;
```
Comment (evantahler) [inline airbyte-integrations/connectors/destination-bigquery/src/test-integration/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGeneratorIntegrationTest.java:25]: lol, that's a lot of imports

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java [hunk 2]
```diff
@@ -395,20 +395,24 @@ String cdcDeletes(final StreamConfig stream,
...
         "quoted_cdc_delete_column", QUOTE + "_ab_cdc_deleted_at" + QUOTE)
     ).replace(
-        // TODO replace `id`, `$.id` with PK
-        // TODO replace `INT64` with PK's type
         """
-        DELETE FROM ${final_table_id} 
-        WHERE 
-          `id` IN (
-            SELECT
-              SAFE_CAST(JSON_VALUE(`_airbyte_data`, '$.id') as INT64) as id
```
Comment (evantahler) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java:411]: 💪 work with the PK array

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`a0b7f170d2`](https://github.com/airbytehq/airbyte/pull/28092/commits/a0b7f170d2fedbc341311a9cd2bbe049c55719a4)) - ✅

⏲️  Total pipeline duration: 25mn13s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_stuff/1689028190/a0b7f170d2fedbc341311a9cd2bbe049c55719a4/destination-bigquery/1.5.2/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery-denormalized test report (commit [`a0b7f170d2`](https://github.com/airbytehq/airbyte/pull/28092/commits/a0b7f170d2fedbc341311a9cd2bbe049c55719a4)) - ✅

⏲️  Total pipeline duration: 17mn24s 

| Step                                                                                         | Result   |
|:---------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery-denormalized/metadata.yaml     | ✅        |
| Connector version semver check                                                               | ✅        |
| QA checks                                                                                    | ✅        |
| Build connector tar                                                                          | ✅        |
| Build destination-bigquery-denormalized docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery-denormalized:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_stuff/1689028190/a0b7f170d2fedbc341311a9cd2bbe049c55719a4/destination-bigquery-denormalized/1.5.2/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery-denormalized test
```

**Pull request merged**